### PR TITLE
Make cilium cni.exclusive configurable

### PR DESCRIPTION
**Pull request by petrutlucian94**
We'll add an annotation, making the cilium "cni.exclusive" setting configurable. This will allow using additional CNIs such as Multus.

Requires: https://github.com/canonical/k8s-snap-api/pull/20
Fixes: https://github.com/canonical/k8s-snap/issues/796

### CODE REVIEW

File: docs/src/snap/reference/annotations.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ the bootstrap configuration.
 | `k8sd/v1alpha1/calico/autodetection-v6/skipInterface` | Enable IP auto-detection based on interfaces that do not match the given regex. | string |
 | `k8sd/v1alpha1/calico/autodetection-v6/canReach` | Enable IP auto-detection based on which source address on the node is used to reach the specified IP or domain. | string |
 | `k8sd/v1alpha1/calico/autodetection-v6/cidrs` | Enable IP auto-detection based on which addresses on the nodes are within one of the provided CIDRs. | []string (comma separated) |
+| `k8sd/v1alpha1/cilium/cni-exclusive` | Make Cilium take ownership over the `/etc/cni/net.d` directory on the node, renaming all non-Cilium CNI configurations to `*.cilium_bak`. This ensures no Pods can be scheduled using other CNI plugins during Cilium agent downtime. Set this to "false" if you wish to use other CNIs such as Multus. | "true"\|"false" |
```
Comment (bschimke95) [inline docs/src/snap/reference/annotations.md:27]: nit: this sounds like you need to set one of the values.
`false`/not set is the default. Please rephrase to something like
```
Set to "true" if you do not want to allow multiple CNIs.
```

File: src/k8s/pkg/k8sd/features/cilium/internal_test.go [hunk 2]
```diff
@@ -119,7 +140,7 @@ func TestInternalConfig(t *testing.T) {
 		{
 			name:           "Nil annotations",
 			annotations:    nil,
-			expectedConfig: config{},
+			expectedConfig: config{cniExclusive: true},
```
Comment (bschimke95) [inline src/k8s/pkg/k8sd/features/cilium/internal_test.go:143]: why do we expect `cniExclusive` here? 
if everything is nil, then we should expect nil and use whatever cilium takes as default.

**Review by bschimke95 — COMMENTED** [review]
Great stuff, added some questions/comments.

File: docs/src/snap/reference/annotations.md (hunk 1 above)
Comment (petrutlucian94) [reply docs/src/snap/reference/annotations.md:27]: Yes, the initial implementation expects one of these values. I'll update it.

**Review by bschimke95 — APPROVED** [review]
LGTM after CI is green

**Comment by petrutlucian94** [comment]
```Error: pkg/k8sd/features/cilium/internal.go:75:48: undefined: apiv1_annotations.AnnotationCNIExclusive```

We need to merge the other PR first and then bump the requirement.

**Pull request merged**